jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -200,6 +185,20 @@ public Builder setCreationTime(Instant creationTime) {
       return this;
     }
 
+    /**
+     * Sets the container entrypoint.
+     *
+     * @param entrypoint the tokenized command to run when the container starts
+     * @return this
+     */
+    public Builder setEntrypoint(@Nullable List<String> entrypoint) {
```

Comment:
What is the point of allowing `null`? Also, I think we should document the behavior: "passing null does nothing; it does not clear the previous value."

Reply:
The point of allowing `null` is to allow for building an image without an entrypoint (since it's an optional field on the image format). This can also be useful for when we propagate base image container configuration so fields like entrypoint could be propagated. But yes, we should defintely document the behavior.

Reply:
Alright, just keep in mind you'd normally consider what locale to use if the text can be something else than English, e.g., https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#toUpperCase(java.util.Locale)

Reply:
If so, why not just clear the entrypoint if `null` is given? That's more intuitive and what I'd expect normally.

Reply:
That makes a lot of sense.

Reply:
Or, actually - I think we should probably have the semantics be like:

`null` - if base image has entrypoint, propagate; otherwise, don't set an entrypoint
non-`null` - set entrypoint to what is given (empty would be empty entrypoint, etc.)